### PR TITLE
Get quantity from request as an integer.

### PR DESCRIPTION
**Pull request by ce-bo**
PR does fix this issue: https://github.com/Sylius/SyliusShopApiPlugin/issues/235

### CODE REVIEW

**Comment by lchrusciel** [comment]
Hey @ce-bo. Thanks for contribution. I would merge it but the tests are failing :( Can you adjust them as well?

**Comment by ce-bo** [comment]
Hi @lchrusciel, the tests have passed now. Could you merge it please?

File: src/Request/ChangeItemQuantityRequest.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function __construct(Request $request)
     {
         $this->token = $request->attributes->get('token');
         $this->id = $request->attributes->get('id');
-        $this->quantity = $request->request->get('quantity');
+        $this->quantity = $request->request->has('quantity') ? $request->request->getInt('quantity') : null;
```
Comment (pamil) [inline src/Request/ChangeItemQuantityRequest.php]: Quantity cannot be null, it's declared as `int`. `$request->request->getInt('quantity', 1)` instead?

File: src/Request/PutOptionBasedConfigurableItemToCartRequest.php [hunk 2]
```diff
@@ -44,7 +44,7 @@ public static function fromArray(array $item)
 
     public static function fromRequest(Request $request)
     {
-        return new self($request->attributes->get('token'), $request->request->get('productCode'), $request->request->get('options'), $request->request->get('quantity'));
+        return new self($request->attributes->get('token'), $request->request->get('productCode'), $request->request->get('options'), $request->request->has('quantity') ? $request->request->getInt('quantity') : null);
```
Comment (pamil) [inline src/Request/PutOptionBasedConfigurableItemToCartRequest.php]: Same here and so on.

File: src/Request/ChangeItemQuantityRequest.php (hunk 1 above)
Comment (lchrusciel) [reply src/Request/ChangeItemQuantityRequest.php]: `$request->request->getInt('quantity', 0)`? This should trigger validation error. Question is, what should be a default behaviour. 1 as a default would be more flexible for frontends...

Comment (pamil) [reply src/Request/ChangeItemQuantityRequest.php]: If we allow not to pass quantity then it makes sense to use 1 as the default value. If we require quantity then we should throw an exception here if it does not exist.

Comment (ce-bo) [reply src/Request/ChangeItemQuantityRequest.php]: The decision is up to you.

Comment (lchrusciel) [reply src/Request/ChangeItemQuantityRequest.php]: Let's add `1` as default. But we need to mention it in docs.

Comment (ce-bo) [reply src/Request/ChangeItemQuantityRequest.php]: Can you adjust it, please?

Comment (lchrusciel) [reply src/Request/ChangeItemQuantityRequest.php]: sure, but it will take some time

**Comment by pjedrzejewski** [comment]
Hi @ce-bo, do you plan to continue on this PR? :)

**Comment by ce-bo** [comment]
Hi @pjedrzejewski, unfortunately not. I already discussed with @lchrusciel that he would rather do it.

**Comment by lchrusciel** [comment]
Thanks @ce-bo for all your effort! And welcome to our community! 🎉

**Comment by ce-bo** [comment]
@lchrusciel thanks for finalizing!

**Comment by lchrusciel** [comment]
I will make a release this week

**Pull request merged**